src/app/campaign/flight/flight.container.spec.ts

```diff
@@ -94,7 +94,8 @@ describe('FlightContainerComponent', () => {
     });
   });
 
-  it('redirects to campaign if the flight does not exist', () => {
+  // TODO: no longer does this because it caused a premature navigation bug, should we create a better way to detect it?
```

Comment:
Yeah, as long as there's a TODO on this, I'm fine with re-introducing this bug.  It's such an edge case anyways.